### ISSUE TITLE
Current progress and problems so far

### ISSUE THREAD

**Issue by quantaji** [quantaji/AMD-SSD#4]
# Current progress
- Wolfpack env finished. (test code [here](https://github.com/quantaji/FoRL-project/blob/main/test_code/test_env.ipynb))
- run a few Adaptive Mechanism Design runs:
    - currently no cooperative evidence on AMD, but there are some cooperation when no AMD applied.
![截屏2023-05-02 21 22 48asdfasd](https://user-images.githubusercontent.com/91256686/235774120-c14572ff-06dd-41fe-a4a0-94d053f86120.png)

**Comment by quantaji** [comment]
# Questions so far
- Is current setting correct? About how to compute awareness. Why is the algorithm not working?
- Is it sensible to pass gradient also through $r_{\text{planner}}$?   $r_{\text{planner}}$ is sampled from central planner's policy, how can we pass gradient through it?
- What batchsize, total time step, learning rate is the common choice?
- Can we use critic, entropy loss, GAE, PPO/TRPO for individual agent to improve convergence?
    - If so, which gradient should be passed to central planner? The actual gradient or the gradient by vanilla loss.
- In the current setting, the actual gradient of individual agent is $\nabla_i \log \pi_i(\tau) (\delta + r_{\text{planner}})$. Is this plausible?
- When we train the critic, do we use the real rewards or r_real+r_planner?
- Is it a good idea to set the prey also have reward?
    - If so, I found that the prey learns to escape and the game turns harder for predator.
- Is it a good idea for predator's policies to share parameter?
